k2/csrc/ragged.h

```diff
@@ -212,6 +216,52 @@ class RaggedShape {
   std::vector<RaggedShapeLayer> layers_;
 };
 
+
+template <typename T, int MAX_DIM>
+struct ArrayAccessor {
```

Comment:
There is an `Array2Accessor` in `array.h`. Shall we also move this one to `array.h`?

Reply:
This is not an array in the same sense, it's a small array accessor (that can be transferred with the lambda).